Assets/Mirror/Core/NetworkBehaviour.cs

```diff
@@ -85,9 +85,9 @@ public abstract class NetworkBehaviour : MonoBehaviour
         // also note that this is a per-NetworkBehaviour flag.
         // another component may not be client authoritative, etc.
         public bool authority =>
-            isClient
-                ? syncDirection == SyncDirection.ClientToServer && isOwned
-                : syncDirection == SyncDirection.ServerToClient;
+            (isClient && syncDirection == SyncDirection.ClientToServer && isOwned)
+            ||
+            (isServer && syncDirection == SyncDirection.ServerToClient);
```

Comment:
I think the change should be this.  The bug is that isClient is true for host client, so the syncDirection as ServerToClient is wrongly false.  By checking IsServer instead, if that's true then isClient being true or false is irrelevant, if isServer is false then isClient will be true for sure, so we don't need to check that separately.
```suggestion
            isServer
                ? syncDirection == SyncDirection.ServerToClient
                : syncDirection == SyncDirection.ClientToServer && isOwned;
```

Reply:
I don't think that the bug is that isClient is true for hosts. My interpretation is that in a host, both isClient and isServer are true intentionally. The values of isClientOnly and isClientServer can be used to tell the difference between host and client-only or server-only instances. If isClient has a bug it should be addressed in a different PR though.

My intention here is to fix the value of authority flag. With the change you suggested, a host would always evaluate as server, and if you checked authority in a client-owned object with client-to-server direction in a host, the value would be false (but it should be true)